pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -397,13 +397,14 @@ func (r *Reconciler) streamLogs(ctx context.Context, o results.Object, logType,
 		return fmt.Errorf("error reading from tkn reader: %w", err)
 	}
 
-	errChanRepeater := make(chan error, 100) //some stuff on the internet says buffered channels are better for GC
+	errChanRepeater := make(chan error, 100)
 
 	// logctx is derived from ctx. Therefore, if ctx is cancelled (either explicitly through a call to its cancel
 	// function or when it reaches its deadline), logctx will be cancelled automatically.
 	logctx, _ := context.WithTimeout(ctx, 10*time.Minute)
 
 	go func(ctx context.Context, echan <-chan error, o metav1.Object) {
+		defer close(errChanRepeater)
```

Comment:
good use of defer @ramessesii2 @sayan-biswas 

where are we closing the original `logChan` and `errChan` from the `reader.Read()`  ?

Reply:
logChan and errChan channels are receiver ends. When the channels ae no longer used, they will be garbage collected.
It's new subject for me, this is my reference - https://stackoverflow.com/questions/8593645/is-it-ok-to-leave-a-channel-open

Reply:
yeah there was a bit of back and forth on that thread @ramessesii2 but yes I think that is the end interpretation.

good find / research @ramessesii2 

that said, given the exchanges I just saw with @pmacik today in slack, I should probably make a more thorough pass of the changes here  .... admittedly I only focused on the mem leak bit initially.  

will do that today

thanks

Reply:
ok @ramessesii2 @sayan-biswas I think I looked at this too quickly yesterday

the close of the `errChanRepeater` should not be in the goroutine that blocks on it !!

move the `defer close(errChanRepeater)` to immediately after the declaration of `errChanRepeater`  in your current line 400

Reply:
The reason why I'm deferring close(errChanRepeater) in this goroutine is because it's important to tell the receiving goroutines that all data have been sent which is here:
```
tknlog.NewWriter(logType, true).Write(&cli.Stream{
		Out: writer,
		Err: writer,
	}, logChan, errChanRepeater)
```
else the above code will not exit the execution. I confirmed this as well while debugging.

Reply:
yep we are good here per the thread below